web/src/labs/marked/parser/Tag.tsx

```diff
@@ -1,6 +1,6 @@
 import { matcher } from "../matcher";
 
-export const TAG_REG = /#([^\s#,]+)/;
+export const TAG_REG = /#((?:[^\s\p{P}]|_)+)/u;
```

Comment:
Consider adding a comment explaining the regular expression. Regular expressions can be hard to understand, and a comment explaining what it does can be very helpful for future maintainers. [medium]